model/event/patch_event.go

```diff
@@ -0,0 +1,52 @@
+package event
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/mongodb/grip"
+)
+
+func init() {
+	registry.AddType(ResourceTypePatch, patchEventFactory)
+	registry.AllowSubscription(ResourceTypePatch, PatchStateChange)
+	registry.RegisterExtraData(ResourceTypePatch, "time-exceeds-n-constant", 0)
+	registry.RegisterExtraData(ResourceTypePatch, "time-exceeds-n-percent", 0.0)
+}
+
+func patchEventFactory() interface{} {
+	return &PatchEventData{}
+}
+
+const (
+	ResourceTypePatch = "PATCH"
+
+	PatchStateChange = "STATE_CHANGE"
```

Comment:
do these types need to be public?

Reply:
Yes, they're used as a part of the event/trigger registries, plus they're used to create events for tests.

It's not possible to combine model/event and the trigger portions of model/notification because it creates a cycle.